File: code/go/sdk/console.go
package datastar

import "fmt"

func (sse *ServerSentEventGenerator) Consolef(mode ConsoleMode, messageFormat string, args ...any) error {
	message := fmt.Sprintf(messageFormat, args...)
	return sse.Console(mode, message)
}

func (sse *ServerSentEventGenerator) Console(mode ConsoleMode, message string, opts ...SSEEventOption) error {
	return sse.Send(
		EventTypeConsole,
		[]string{string(mode) + " " + message},
		opts...,
	)
}

File: code/ts/sdk/src/console.ts
import { requireThatString, assertThatString } from "@cowwoc/requirements";
import { DatastarEvent, DatastarEventOptions } from "./types.ts";

export const consoleModes = [
    "assert"
    , "clear"
    , "count"
    , "countReset"
    , "debug"
    , "dir"
    , "dirxml"
    , "error"
    , "group"
    , "groupCollapsed"
    , "groupEnd"
    , "info"
    , "log"
    , "table"
    , "time"
    , "timeEnd"
    , "timeLog"
    , "trace"
    , "warn"
] as const;

export type ConsoleMode = typeof consoleModes[number];

interface ConsoleGroupEndEvent extends DatastarEvent {
    type: "datastar-console";
    mode: "groupEnd";
}

interface ConsoleMessageEvent extends DatastarEvent {
    type: "datastar-console";
    mode: Omit<ConsoleMode, "groupEnd">;
    message: String;
};

export type ConsoleEvent = ConsoleGroupEndEvent | ConsoleMessageEvent;